### PR TITLE
Add color picker for Design Mode icons

### PR DESCRIPTION
**Pull request by joshlory**
Users can now choose a fill color for Design Mode icons:

### CODE REVIEW

**Comment by davidsbailey** [comment]
Hey Josh, I'd be happy to look at this, however I'm under the gun to finish something this afternoon and on vacation tomorrow, so I may not be able to give it a thorough look until Monday morning.

File: apps/src/applab/designElements/button.jsx [hunk 1]
```diff
@@ -23,6 +24,24 @@ var ButtonProperties = React.createClass({
   render: function () {
     var element = this.props.element;
 
+    var iconColorPicker;
+    var canonicalImage = element.getAttribute('data-canonical-image-url');
+    if (ICON_PREFIX_REGEX.test(canonicalImage)) {
+
```
Comment (Bjvanminnen) [inline apps/src/applab/designElements/button.jsx]: nit: extra line

File: apps/src/applab/designElements/button.jsx [hunk 2]
```diff
@@ -23,6 +24,24 @@ var ButtonProperties = React.createClass({
...
+      var handleIconColorChange = function (value) {
+        this.props.handleChange('icon-color', value);
+        this.props.handleChange('image',
+          element.getAttribute('data-canonical-image-url'));
+      }.bind(this);
```
Comment (Bjvanminnen) [inline apps/src/applab/designElements/button.jsx]: any reason not to have `handleIconColorChange` live on the class?

File: apps/src/applab/designElements/image.jsx [hunk 3]
```diff
@@ -20,6 +21,24 @@ var ImageProperties = React.createClass({
   render: function () {
     var element = this.props.element;
 
+    var iconColorPicker;
+    var canonicalImage = element.getAttribute('data-canonical-image-url');
+    if (ICON_PREFIX_REGEX.test(canonicalImage)) {
+
```
Comment (Bjvanminnen) [inline apps/src/applab/designElements/image.jsx]: nit: extra line

File: apps/src/applab/designElements/button.jsx (hunk 1 above)
Comment (Bjvanminnen) [inline apps/src/applab/designElements/button.jsx]: same in other jsx files

File: apps/src/applab/designElements/button.jsx (hunk 2 above)
Comment (Bjvanminnen) [inline apps/src/applab/designElements/button.jsx]: same in other jsx files

File: apps/src/applab/designMode.js [hunk 4]
```diff
@@ -180,13 +180,14 @@ designMode.onPropertyChange = function(element, name, value) {
  * @param value {string} An icon identifier of the format "icon://fa-icon-name".
  * @return {string}
  */
-function renderIconToString(value) {
+function renderIconToString(value, element) {
```
Comment (Bjvanminnen) [inline apps/src/applab/designMode.js:184]: add @param for element

**Comment by Bjvanminnen** [comment]
couple comments, but otherwise lgtm :)

**Pull request merged**